lib/sober_swag/parser.rb

```diff
@@ -2,12 +2,18 @@ module SoberSwag
   ##
   # Parses a *Dry-Types Schema* into a set of nodes we can use to compile.
   # This is mostly because the vistior pattern sucks and catamorphisms are nice.
```

Comment:
```suggestion
  # This is mostly because the visitor pattern sucks and catamorphisms are nice.
```

Also, what is "visitor pattern"?

Reply:
https://en.wikipedia.org/wiki/Visitor_pattern